src/extension/infer.rs

```diff
@@ -312,6 +307,14 @@ impl UnificationContext {
             match node_type.signature() {
                 // Input extensions are open
                 None => {
+                    if node == hugr.root()
```

Comment:
The check on this line is equivalent to the one removed above (lines 271-275); the `||` on next line extends it and is a first stab at what nodes we'd expect to be unconnected. (An alternative might be to skip creating `m_input` and `m_output` earlier in the loop...that might be better, happy to give that a go). Note #457 .

This still won't address random unconnected bits of graph, which is to say, malformed graphs, tho. (Only really an issue for tests of validation - some of these have to jump through hoops to get the error they expect rather than "CantInfer")